### ISSUE TITLE
Serializer crashes on blocked definitions when using --allow-unsolved-metas

### ISSUE THREAD

**Issue by quomat** [agda/agda#7044]
During tinkering with `agda-unimath` repo (and in the process, resetting all definitions to holes), I've encountered an "IMPOSSIBLE" bug.

I reduced the error to a few files: [impossible.zip](https://github.com/agda/agda/files/13802686/impossible.zip)

## Reproduction

From the `src` directory, after extracting the attached zip file, invoke `agda .\foundation\action-on-identifications-functions.lagda.md --allow-unsolved-metas`.

## Expectation

Agda checks the files successfully.

## Result

An internal error has occurred. Please report this as a bug.
Location of the error: __IMPOSSIBLE__, called at src/full\Agda\TypeChecking\Serialise\Instances\Internal.hs:232:21 in Agda-2.6.2.2-I0mdgVgXQQ9I75Kwxokerz:Agda.TypeChecking.Serialise.Instances.Internal

I am able to reproduce this both in native Windows and WSL, so it seems independent of the OS.

**Comment by sattlerc** [comment]
Simplified example:

```agda
{-# OPTIONS --allow-unsolved-metas #-}
module Inner where

data T : Set where
  t : T

postulate
  X : Set
  x : X

A : Set
A = ?

foo : A → X
foo t = x
```

```agda
module Outer where

import Inner
```

The `IMPOSSIBLE` in `Agda.TypeChecking.Serialise.Instances.Internal` is this:
```haskell
instance EmbPrj Blocked_ where
  icod_ (NotBlocked a b) = icodeN' NotBlocked a b
  icod_ Blocked{} = __IMPOSSIBLE__

  value = valueN NotBlocked
```

**Comment by sattlerc** [comment]
The bug only shows when generating an interface file with open metas. This apparently doesn't happen when checking the main module, which has the unfortunate effect of masking the bug in the absence of the above import indirection. 

The strategy for `--allow-unsolved-metas` is to turn open metas into postulates. In this case, when generating the interface file for `Inner`, the check of the definition of `foo : A → X` is blocked on the metavariable for `A`. I do not understand how we can turn `A` into a postulate without dropping `foo` from the interface file.

Relevant code in `Agda.Interaction.Imports`:
```haskell
      whenM (optAllowUnsolved <$> pragmaOptions) $ do
        reportSLn "import.iface.create" 7 "Turning unsolved metas (if any) into postulates."
        withCurrentModule (scope ^. scopeCurrent) openMetasToPostulates
        -- Clear constraints as they might refer to what
        -- they think are open metas.
        stAwakeConstraints    `setTCLens` []
        stSleepingConstraints `setTCLens` []
```

Constraints after turning open metas into postulates, showing the blocked constraint that gets dropped:
```
Current constraints
  awake 
  asleep Check definition of foo : A → X
           stuck because
             /home/noname/agda-7044/agda-unimath/src/Inner.agda:15,5-6
             Cannot split on argument of unresolved type A
             when checking that the pattern t has type A
           (blocked on _1)
```

**Comment by sattlerc** [comment]
PS: When printing the signatures of the interface, I see `Inner.hcomp-T`. Why is that generated without `--cubical-compatible`?

**Comment by andreasabel** [comment]
This works correctly with Agda 2.5.4 and 2.6.0 but triggers an internal error starting with 2.6.1.

Blocked definitions are created here: https://github.com/agda/agda/blob/ea16c4cf0833afa2b2386260928b804d7f471112/src/full/Agda/TypeChecking/Rules/Def.hs#L133-L137

**Comment by andreasabel** [comment]
Bisection suggest that the fix b84b215fa89ad36bf9251d7b4f03ad1a2dbd5419 for #4255 introduced the current behavior. (CC @jespercockx)
- #4255
 
> Date:   Mon Dec 30 17:18:03 2019 +0100
> [ fix #4255 ] Keep track of metavariable blocking postponed function definition

**Comment by sattlerc** [comment]
What is the intended semantics of serializing a definition that's blocked on a meta (assuming `--allow-unsolved-metas`)?

**Comment by andreasabel** [comment]
I suppose we should turn it into a postulate, because it can never be unblocked...

Update: I suppose it will internally already be a postulate, since checking the definition (and converting it into internal syntax) never succeeded.
